src/inference/src/dev/core_impl.cpp

```diff
@@ -271,22 +262,23 @@ ov::SoPtr<ov::ICompiledModel> ov::CoreImpl::compile_model(const std::shared_ptr<
     }
     auto plugin = get_plugin(parsed._deviceName);
     ov::SoPtr<ov::ICompiledModel> res;
-    auto cacheManager =
-        coreConfig.get_cache_config_for_device(parsed._deviceName, device_supports_cache_dir(plugin), parsed._config)
-            ._cacheManager;
+    property_manager.merge_property(parsed._config, plugin.get_name());
+    auto cacheManager = property_manager.get_property(ov::cache_manager.name(), plugin.get_name())
+                            .as<std::shared_ptr<ov::ICacheManager>>();
     auto cacheContent = CacheContent{cacheManager};
+    ov::AnyMap _config = property_manager.exclude_property(parsed._config);
     if (!forceDisableCache && cacheManager && device_supports_import_export(plugin)) {
```

Comment:
do you think we can remove internal config key `FORCE_DISABLE_CACHE` in a separate PR?
It's currently used in HETERO to prevent double caching:
- HETERO caches the whole model with subgraphs
- It passes `FORCE_DISABLE_CACHE` to disable caching of specific subgraphs, because the whole model is already cached. I suppose we can use inside the HETERO for each subgraph:
```c++
GetCore()->compile_model(subgraph[i], device, ov::cache_dir("")); // ov::cache_dir("") explicitly disable cache
```